include/revng/Model/Type.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/ADT/RecursiveCoroutine.h"
+#include "revng/ADT/UpcastablePointer.h"
+
+/* TUPLE-TREE-YAML
+name: Type
+doc: Base class of model types used for LLVM-style RTTI
+type: struct
+fields:
+  - name: Kind
+    type: TypeKind
+  - name: IsConst
+    type: bool
+    optional: true
+abstract: true
+key:
+- Kind
```

Comment:
Mmmh, this is a bit dangerous. If you put this in a `SortedVector`, it will collapse all the elements by kind.

Can we leave the key out entirely?
If not, leave a warning not to use `Type` in a `SortedVector`.

Reply:
Totally agreed. I planned to explicitly disable KOTs or something.

But it's necessary. Sadly, if `Kind` is not a part of the key, [de-]serialization breaks.

Reply:
I managed to work around this - no key necessary :tada: